lib/msg-format.c

```diff
@@ -58,6 +59,19 @@ msg_format_postprocess_message(MsgFormatOptions *options, LogMessage *msg)
                              time_zone_info_get_offset(options->recv_time_zone_info,
                                                        msg->timestamps[LM_TS_RECVD].ut_sec));
     }
+  if (G_UNLIKELY(options->flags & LP_NO_MULTI_LINE))
```

Comment:
I think it makes better sense that we apply flags to both parsed/unparsed/error cases. If we asked syslog-ng not to give us newlines in messages, that's possibly because we  have something in our processing pipeline that doesn't handle that.

If in rare cases we don't perform as asked, then that's not that intuitive.

So I think it's better to fix this behavior.